hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -79,6 +89,37 @@ def has_manufacturing_errors(self) -> list:
 
         return []
 
+    def validate_gateway_transaction_if_present(self) -> list:
+        """
+        Check if add gateway transaction data is present in this report and
+        that it's valid.
+
+        returns list:
+            - If empty it means either there was no transaction data
+            or it was valid.
+            - In case of missing ADD_GATEWAY_TXN_PAYLOAD_KEY
+            the ADD_GATEWAY_TXN_PAYLOAD_KEY is returned.
+            - If payload is found but has missing or empty keys those
+            keys are returned.
+        """
+        errors = []
+        if self.ADD_GATEWAY_TXN_NAME_KEY in self.keys():
+            if self.ADD_GATEWAY_TXN_PAYLOAD_KEY not in self.keys():
```

Comment:
I would take a different approach to this. First, I assume but should have been explicit, that https://github.com/NebraLtd/hm-diag/issues/283 should add the needed fields through the creation of a DestinationAddGatewayTxnDiagnostic (or similar) @SebastianMaj. That Diagnostic should be responsible for adding to the DiagnosticReport and appending an error if necessary.

Then from HPT, the new logic would be:
- If a diagnostics_report has_key ADD_GATEWAY_TXN_NAME_KEY and ADD_GATEWAY_TXN_PAYLOAD_KEY in diagnostics_report.errors => retry
- Else success

We don't want DiagnosticsReport to have to know how to validate any arbitrary Diagnostic that another app (eg hm-diag) implements.

Reply:
> That Diagnostic should be responsible for adding to the DiagnosticReport and appending an error if necessary.

Yes if that is being done in hm-diag then it won't be needed here.

I guess we should have had a group call with all people involved in this epic to clearly sketch out what needs to be done where.

Reply:
Also moved out of DiagnosticsReport and will be handled at HPT end.